micronaut-maven-plugin/src/main/java/io/micronaut/maven/testresources/DefaultServerFactory.java

```diff
@@ -75,8 +79,17 @@ public void startServer(ServerUtils.ProcessParameters processParameters) {
         processParameters.getSystemProperties().forEach((key, value) -> cli.add("-D" + key + "=" + value));
         cli.add("-cp");
         cli.add(processParameters.getClasspath().stream().map(File::getAbsolutePath).collect(Collectors.joining(File.pathSeparator)));
+        String mainClass = processParameters.getMainClass();
+        if (mainClass == null) {
+            throw new IllegalStateException("Main class is not set");
+        }
         cli.add(processParameters.getMainClass());
         cli.addAll(processParameters.getArguments());
+
```

Comment:
Don't think any other parameter besides `javaBin` and `mainClass` can be null because all others are added as `cli.addAll()` and it would fail if param is null. Added debug logging command params, maybe if something slipped as null could be noticed.

Reply:
Not looking useful, looks like logging is always disabled in `SystemStreamLog` that is actual logger implementation: 
```
public boolean isDebugEnabled() {
        // TODO Not sure how best to set these for this implementation...
        return false;
    }

    /**
     * @see org.apache.maven.plugin.logging.Log#isInfoEnabled()
     */
    public boolean isInfoEnabled() {
        return true;
    }

    /**
     * @see org.apache.maven.plugin.logging.Log#isWarnEnabled()
     */
    public boolean isWarnEnabled() {
        return true;
    }

    /**
     * @see org.apache.maven.plugin.logging.Log#isErrorEnabled()
     */
    public boolean isErrorEnabled() {
        return true;
    }
```